### PR TITLE
Migrate documentation to Puppet strings

### PR DESCRIPTION
**Pull request by nod0n**
#### Pull Request (PR) description
Replace class and define documentation with puppet strings type documentation.

### CODE REVIEW

**Review by oranenj — APPROVED** [review]
I'm not going to read 100% through all of this, but since it's only documentation changes and I can't spot anything wrong, I'll trust that it's fine.

File: docs/_index.html [hunk 1]
```diff
@@ -0,0 +1,142 @@
+<!DOCTYPE html>
```
Comment (ekohl) [inline docs/_index.html:1]: We don't generate the HTML documentation, only the reference. There is `rake strings:generate:reference` just for that.

**Review by ekohl — CHANGES_REQUESTED** [review]

File: docs/_index.html (hunk 1 above)
Comment (nod0n) [reply docs/_index.html]: removed: https://github.com/voxpupuli/puppet-sftp_jail/pull/70/commits/7df254831960a96013ab8a32c9ef92c9a81c9824

File: manifests/jail.pp [hunk 2]
```diff
@@ -1,3 +1,49 @@
...
+#   giving the user a place to land. Also sets group ownership for `/incoming`.
+#
+# @param match_group
+#   Set the group that SSHd will look for when redirecting users to the jail.
+#   Useful for shared jails. Defaults to the value of `group`.
+#
+# @param password_authentication
+#   OpenSSH Password Authentication
+#
 define sftp_jail::jail (
```
Comment (ekohl) [inline manifests/jail.pp:47]: This could use a `@summary` tag

**Review by ekohl — APPROVED** [review]
This should be squashed. We can do it on merge, but we shouldn't forget that.

Do you want to add the `@summary` tag before we merge?

File: manifests/jail.pp (hunk 2 above)
Comment (nod0n) [reply manifests/jail.pp:52]: done (and squashed)

Comment (ekohl) [reply manifests/jail.pp:52]: Would you mind regenerating REFERENCE.md so it shows the summary tag that was added? Other than that :+1:

**Review by igalic — APPROVED** [review]
👍🏻

File: REFERENCE.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+## Table of Contents
+
+### Classes
+
+* [`sftp_jail`](#sftp_jail): Manage SFTP Jails
+
+### Defined types
+
+* [`sftp_jail::jail`](#sftp_jailjail)
```
Comment (ekohl) [inline REFERENCE.md:13]: If you regenerate, this should have a summary as well.

**Review by ekohl — APPROVED** [review]
Thanks!

**Pull request merged**